src/System.Security.Cryptography.OpenSsl/src/System/Security/Cryptography/ECDsaOpenSsl.ImportExport.cs

```diff
@@ -0,0 +1,270 @@
+                curve.CurveType,
+                parameters.Q.X, parameters.Q.X.Length,
+                parameters.Q.Y, parameters.Q.Y.Length,
+                includePrivate ? parameters.D : null,
+                includePrivate ? parameters.D.Length : 0,
+                curve.P, curve.P.Length,
+                curve.A, curve.A.Length,
+                curve.B, curve.B.Length,
+                curve.G.X, curve.G.X.Length,
+                curve.G.Y, curve.G.Y.Length,
+                curve.N, curve.N.Length,
+                curve.H, curve.H.Length,
+                curve.S, curve.S == null ? 0 : curve.S.Length);
+
+            if (key == null || key.IsInvalid)
+                throw Interop.Crypto.CreateOpenSslCryptographicException();
+
+            return key;
+        }
+
+        private static SafeEcKeyHandle ImportCharacteristic2CurveParameters(ECParameters parameters, ECCharacteristic2Curve curve, bool includePrivate)
+        {
+            //TODO byte[] m = new byte[1] { (byte)curve.m };
+            SafeEcKeyHandle key = Interop.Crypto.EcKeyCreateByExplicitParameters(
+                curve.CurveType,
+                parameters.Q.X, parameters.Q.X.Length,
+                parameters.Q.Y, parameters.Q.Y.Length,
+                parameters.D, parameters.D == null ? 0 : parameters.D.Length,
+                curve.M, curve.M.Length,
+                curve.A, curve.A.Length,
+                curve.B, curve.B.Length,
+                curve.G.X, curve.G.X.Length,
+                curve.G.Y, curve.G.Y.Length,
+                curve.N, curve.N.Length,
+                curve.H, curve.H.Length,
+                curve.S, curve.S == null ? 0 : curve.S.Length);
+
+            if (key == null || key.IsInvalid)
+                throw Interop.Crypto.CreateOpenSslCryptographicException();
+
+            return key;
+        }
+
+        private static void CheckInvalidKey(SafeEcKeyHandle key)
+        {
+            if (key == null || key.IsInvalid)
+            {
+                throw new CryptographicException(SR.Cryptography_OpenInvalidHandle);
+            }
+        }
+
+        // Does the public key have values
+        private static bool IsPublicKeyDefined(ECParameters parameters)
+        {
+            return (
+                parameters.Q != null &&
+                parameters.Q.X != null &&
+                parameters.Q.X.Length > 0 &&
+                parameters.Q.Y != null &&
+                parameters.Q.Y.Length > 0);
+        }
+
+        private static bool ValidateNamedCurve(ECNamedCurve curve)
+        {
+            if (curve.Oid == null ||
+                string.IsNullOrEmpty(curve.Oid.FriendlyName))
+                return false;
+
+            return true;
+        }
+
+        private static bool ValidateExplicitCurve(ECExplicitCurve curve)
+        {
+            // todo verify P and M
+            if (curve.A == null || // length can be 0 on some curves (sect283k1)
```

Comment:
I'll add this to discussion list. This is really a result of openssl bignum. It really is not an optional parameter - the value is zero.